### ISSUE TITLE
[Feature Suggest] Image category, tags, comment, share and export

### ISSUE THREAD

**Issue by Dubidubiduu** [nextcloud/maps#181]
Hey everyone,

so I will briefly explain my use case: I'm a politician who was told to get some places where the town should put some bicycle stands. To discuss this in our little body it would be great to not only have the coordinates, but also the streets and a picture from that very location. It would also be very cool to have all the information in a map, which I could share to the public and also to get some ppl. to add their ideas directly (write access).

- So, I would love to have some taxonomy for images, categories or tags
- Add comment/description to image
- Would be awesome to share maps by these taxonomies. Also with write access.
- Also would love to export a taxonomy to a html - table. Export options: coordinates, comment, map-tile, image

Great work thou!!

**Comment by jancborchardt** [comment]
Right, so this could already be done right now with the ability to show pictures on the map, and the ability to show favorites which also have a detail description field.

The only thing that’s missing to achieve what you need seems to be **Share favorite locations #70** :)

**Comment by Dubidubiduu** [comment]
If I understood you correctly that means to create a favorite and a image for the corner I want to add my bicycle stand, right? Mhm, seems possible but not the best work-flow in general, cuz these are one dataset for me.

Also I not only want to share the favorite location, but also the attached image.. Isn't it a thing to think about to attach images to favorites?

Thanks anyway!

**Comment by jancborchardt** [comment]
@Dubidubiduu as this is a very very specific request and the app is still very new naturally it might take a bit longer to fit the pieces together. And we will also focus on essential things first. :)

> Isn't it a thing to think about to attach images to favorites?

@eneiluj @tacruc is this a standardized thing at all?

**Comment by daniel-callejas-sevilla** [comment]
Categorisation of things on the map is an essential feature which is currently very limited. I feel the title of this issue hits the nail on the head, even if the use case presented seems to be a narrow one.

I would rewrite this issue as follows:

- Everything that gets added to the map can have tags (zero to many). This applies to favorites, images, GPS tracks, and any new features that this add-on allows in the future.
- Tags are general (as opposed to current "Categories" sorting which is specific only to points on map) — example: the tag "2019 holiday" can be applied to favorites on the map, to images and videos, and to GPS tracks.
- Visualization on the map should allow me to show "all features with a given tag". For example, I would click on "2019 holiday" and this would display all position markers, all pictures, and all GPS tracks having that tag.
- Tags need not be specific to the nextcloud-maps add-on. For example, image files already support a list of tags. These should work both ways (adding a tag to a picture in nextcloud-maps could update the original image file, and adding a tag in an image file with an external app results in nextcloud-maps using it for categorization).
- Later on, supporting combinations of tags in the display would be great. For example: search all features with tags "2019 holiday" and "cool spots" would narrow down the map to that awesome hidden place, with both a marker for its position, the GPS track with the path followed to find it, and the pictures clicked there. This will lead to a search syntax.
- It should be possible to export "by tag" or a more advanced "export search results as". The most natural format for this seems to be a GeoJSON file with markers, GPS tracks, and links to images/videos.
- Of course items without any tag remain useable (for people who don't need or want classification features).
- Routes should be able to get saved and tagged too.

Disclaimer: I spent some time lately working on a standalone webapp that tries to do all of the above, because I was not aware that nextcloud-maps existed. Right now I'm trying to figure out if I want to to continue my work in parallel (pure client-side JS app using any WebDAV server for storage, same spirit as the Joplin notetaking app) or join this effort (restricted to Nexcloud).

**Comment by daniel-callejas-sevilla** [comment]
Addendum to my earlier brick of words: in my view, the central concept of nextcloud-maps should be "stuff that has geographical coordinates and a set of tags". Whether a feature is a point, an image, a video, a GPS track, or something else altogether is just a minor consideration for display purposes (although it may be useful to have a tag for that, to narrow down searches eg. find me GPS tracks only).

Extra ideas:
- Link the [Nextcloud Joplin app](https://github.com/laurent22/joplin-nextcloud/) and nextcloud-maps, using the same set of tags in both apps. Then show on map any Joplin notes that have a geo: location in them (again, using the note's tags to determine visibility), and allow quickly to create a geotagged Joplin note from a click on the map.
- Provide a "table view" that gives an alternate to browse items in tabular form, instead of spread on the map (in the same spirit as overpass-turbo search results).

**Comment by jancborchardt** [comment]
Just as FYI:
- We are at an early stage in the Maps app, so we are focusing on the core things for now.
- This is quite a bit of work and we can’t sink time into stuff like that at the moment.
- Our reference is and will be Google Maps, consumer-facing. Not people who want to browse stuff in tabular form.

**Comment by kadarpik** [comment]
in street phootgraphy or installation service company, it would be nice to tag problems or subjects in nextcloud files module and filter those in maps. I have 20000+ photos in maps and only the limit is zoom and time for filtering, tags might help a lot. I somone shoots a trouble photo and uploads to NC and adding trouble tag, it is easy to find trouble sites near by,

**Comment by Spartachetto** [comment]
Just to connect the related issues. I described maps as the possible right solution for a filtering problem posed in a photos issue (https://github.com/nextcloud/photos/issues/445#issuecomment-719901619).

It seems to me that the functionalities asked here would be the right ones to answer that need too
